src/playwrightTest.ts

```diff
@@ -45,31 +46,31 @@ export class PlaywrightTest {
   }
 
   public async listTests(config: PlaywrightTestConfig, projectName: string, fileOrFolder: string): Promise<playwrightTestTypes.JSONReport | null> {
-    const proc = await this._executePlaywrightTestCommand(config, projectName, ['--list', escapeRegExp(fileOrFolder)]);
+    const jsonOutputPath = path.join(os.tmpdir(), createGuid());
+    const proc = await this._executePlaywrightTestCommand(jsonOutputPath, config, projectName, ['--list', escapeRegExp(fileOrFolder)]);
     if (proc.code !== 0) {
       if (proc.stderr.includes('no tests found.'))
         return null;
       throw new Error(proc.stderr || proc.stdout);
     }
     try {
-      return JSON.parse(proc.stdout);
-    } catch (error) {
-      logger.debug('could not parse JSON', proc.stdout, proc.stderr);
-      throw error;
+      return JSON.parse(await fs.promises.readFile(jsonOutputPath, 'utf8'));
+    } finally {
+      await fs.promises.unlink(jsonOutputPath);
     }
   }
 
-  public async runTest(config: PlaywrightTestConfig, projectName: string, path: string, line: number): Promise<playwrightTestTypes.JSONReport> {
-    const proc = await this._executePlaywrightTestCommand(config, projectName, [`${escapeRegExp(path)}:${line}`]);
+  public async runTest(config: PlaywrightTestConfig, projectName: string, testPath: string, line: number): Promise<playwrightTestTypes.JSONReport> {
+    const jsonOutputPath = path.join(os.tmpdir(), createGuid());
```

Comment:
let's prefix this file with something related to the name of this project:

```js
const jsonOutputPath = path.join(os.tmpdir(), 'playwright-vscode-test-exetension-' + createGuid());
```

this way it'll be possible to trace back some leftovers if any.